src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/PledgeService.cs

```diff
@@ -76,5 +79,36 @@ public async Task<GetApplicationsResponse> GetApplications(long accountId, int p
 
             return JsonConvert.DeserializeObject<GetApplicationsResponse>(await response.Content.ReadAsStringAsync());
         }
+
+        public async Task<GetApplicationResponse> GetApplication(long accountId, int pledgeId, int applicationId, CancellationToken cancellationToken = default)
+        {
+            var response = await _client.GetAsync($"accounts/{accountId}/pledges/{pledgeId}/applications/{applicationId}", cancellationToken);
+            GetApplicationResponse applicationResponse = null;
+
+            if (response.IsSuccessStatusCode)
+            {
+                applicationResponse = JsonConvert.DeserializeObject<GetApplicationResponse>(await response.Content.ReadAsStringAsync());
+            }
+            else if (response.StatusCode != HttpStatusCode.NotFound)
```

Comment:
I presume this else part is to throw the exception for a non successful operation back up the stack. if that's the case it would be cleaner to return the application response in the if and remove the else part and just call EnsureSuccessStatusCode.

Reply:
Think this was added earlier and from which I have branched.